src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/buildTransitive/net6.0/Microsoft.Extensions.Telemetry.Abstractions.targets

```diff
@@ -1,11 +1,11 @@
 <Project>
   <!-- This package should replace the Microsoft.Extensions.Logging.Abstractions source generator. -->
-  <Target Name="_Microsoft_Extensions_Logging_AbstractionsRemoveAnalyzers" 
+  <Target Name="_Microsoft_Extensions_Logging_AbstractionsRemoveAnalyzers"
           Condition="'$(DisableMicrosoftExtensionsLoggingSourceGenerator)' == 'true'"
-          AfterTargets="ResolveReferences">
+          BeforeTargets="CoreCompile">
     <ItemGroup>
-      <_Microsoft_Extensions_Logging_AbstractionsAnalyzer Include="@(Analyzer)" Condition="'%(Analyzer.AssemblyName)' == 'Microsoft.Extensions.Logging.Generators' Or
-                                                                                           '%(Analyzer.NuGetPackageId)' == 'Microsoft.Extensions.Logging.Abstractions'" />
+      <_Microsoft_Extensions_Logging_AbstractionsAnalyzer Include="@(Analyzer)" Condition="'%(Analyzer.Filename)' == 'Microsoft.Extensions.Logging.Generators' Or
```

Comment:
Not sure I'm following this. The analyzer will always be called Microsoft.Extensions.Logging.Generators, right? When will it be called Microsoft.Extensions.Logging.Abstractions?

Reply:
The analyzer (source-generator) is shipped as part of the `Microsoft.Extensions.Logging.Abstractions` package. You can check the line below to ensure that: https://github.com/dotnet/extensions/blob/75a0edec8499be9c348b708bf3730a81d93d5e15/src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/Microsoft.Extensions.Telemetry.Abstractions.csproj#L44

Reply:
Good catch. @RussKie, should we remove the second part of the condition?

Reply:
Tbh, I don't understand why the original condition was written the way it was - that's a question to @joperezr.

I used `%(Analyzer.Filename)` because it is a) more obvious and b) this is what I saw inspecting binlogs.

If there's no real reason/need for the second condition - I'd remove it.

Reply:
I decided to get back to the original approach, as it makes sense to check the NuGet package ID. But please let me know if you have any doubts